### ISSUE TITLE
Consider disallowing `allowMultipleArguments`

### ISSUE THREAD

**Issue by mikeselander** [humanmade/coding-standards#128]
In https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/commit/bb8a48671e213a5588a6439ea52411eeefab4b0f WPCS disallowed multiple arguments on a single line when using a multi-line function signature. I restored the default behavior to avoid throwing a bunch of new flags without wider discussion among the team.

So, the question now is whether or not we should follow WPCS' lead in disallowing this behavior or continue on as we have with allowing it.

Ref: #127

**Comment by rmccue** [comment]
Can you clarify what exactly this rule is? 🤔

**Comment by mikeselander** [comment]
The rule is `PEAR.Functions.FunctionCallSignature.MultipleArguments`. 

> Only one argument is allowed per line in a multi-line function call

See: https://github.com/squizlabs/PHP_CodeSniffer/blob/bff45d96700e97711b49c221377cc238be498b16/src/Standards/PEAR/Sniffs/Functions/FunctionCallSignatureSniff.php#L554-L576

**Comment by rmccue** [comment]
Right, but specifically, what would this affect? e.g. does it affect the following:

```
do_something( [
    'foo' => 'bar',
] );
```

Or

```
do_something( 'foo', [
    'foo' => 'bar',
] );
```

Both of those are relatively common patterns in our code.

**Comment by mikeselander** [comment]
Well, for example the following from our tests is flagged and would be disallowed. Your second example would definitely get flagged, I _think_ (have not yet tested) that your first example might pass.

https://github.com/humanmade/coding-standards/blob/master/tests/fixtures/pass/use-order.php#L38-L40

This is a very common pattern in our code, which is why I clawed it back in #127.

**Comment by kadamwhite** [comment]
While this is in principle a good idea, getting the behaviour that @rmccue , @mikeselander and I agreed we would want might have required writing a custom rule, which feels too complex.
